String_Lab.js

```diff
@@ -0,0 +1,184 @@
+console.log("---------------------------------- \n");
+
+
+// 7. Create a withoutLast code block that takes a single string variable,
+//and returns a copy of the string without the last character
+console.log(`Removing the last character from the string : ${string}`);
+console.log("Methode1 : " + string.slice(0,string.length - 1));
+
+let withoutLast = "";
+for (let i = 0; i < string.length - 1; i++){
+  withoutLast += string[i];
+}
+console.log(`Methode2 : ${withoutLast}`);
+console.log(`Methode3 : ${string}\b `);
+
+//let last = string[string.length - 1];
+//console.log("Methode4 : ", string.split(last));
+
+let str3 = "Had some hard time with this part"
+console.log(`\nRemoving the last word from the string : ${str3}`);
+let space = str3.lastIndexOf(" ");
+
+if (space === -1){
+  console.log(`${str3} is a one word string, this operation wont affect it`);
+} else {
+  withoutLast = str3.slice(0, space);
+  console.log(withoutLast);
+}
+console.log("---------------------------------- \n");
+
+
+// Bonus1:
+//Write a code block that adds a "#" to a string if the string has an odd
+//number of characters, otherwise leave it as is. print/log the result
+//Examples:
+//let name = 'taq'  // should log 'taq#'
+//let name2 ='john' // should log 'john'
+
+let name = "Taq"  // should log 'taq#' name2 ='john' // should log 'john'
+
+if (name.length % 2 === 1){
+  console.log(`${name} has ${name.length} characters (odd number) so it will become: `);
+  name += "#";
+  console.log(name);
+} else console.log(`${name} has ${name.length} characters (eveb number) so it wont change:\n${name}`);
+console.log("---------------------------------- \n");
+
+
+//Bonus2:
+//Write a code block that prints/logs the 3rd to last characters of a string.
+//If the string is not at least 3 characters long then log an error message.
+let sentence = "Doing something", word = "";
+
+if (sentence.length >= 3){
+  word = sentence.slice(-3)
+  console.log(`The 3 last characters of ${sentence} are ${word}`);
+} else console.log(`Error : ${sentence} is too short for this operation`);
+console.log("---------------------------------- \n");
+
+
+//Bonus3:
+//Write a code block that prints/logs whether a string x starts with a vowel.
+
+let x = "Engine";
+
+switch (x[0].toLowerCase()){
+  case "a":
+  case "e":
+  case "i":
+  case "o":
+  case "u":
+  case "y":
+    console.log(`${x} starts with the vowel ${x[0]}`);
+    break;
+  default :
+    console.log(`${x} doesn't start with a vowel`);
+}
+console.log("---------------------------------- \n");
+
+
+//Bonus4:
+//Write a code block that given two strings, str1 and str2,
+//prints/logs a new string with str2 mashed into the middle of str1
+//Examples:
+//For str1 = 'help' and str2 = 'me'. Your program should log 'hemelp'
+//For str1 = 'hello' and str2 = 'world'. Your program should log 'helworldlo'
+
+let str1 = "hello", str2 = "world"; //Your program should log 'hemelp'
+let part1 = str1.slice(0, Math.ceil(str1.length / 2));
```

Comment:
Nice, now you know `Math.ceil` and `Math.floor`. Clever application here specially of `Math.ceil`

Reply:
Thanks a lot, appreciate all your feedback